pandas/core/missing.py

```diff
@@ -478,6 +500,66 @@ def _akima_interpolate(xi, yi, x, der=0, axis=0):
         return [P(x, nu) for nu in der]
 
 
+def interpolate_1d_fill(
```

Comment:
can you move this under interpolate_1d

Reply:
`interpolate_1d_fill` is a special case needed by `blocks._interpolate_with_fill()` to support the limit kwargs with method `pad`. As you requested in the comment above `interpolate_1d_fill` is now called from within `interpolate_2d` which is what `blocks._interpolate_with_fill()` calls.

You request, moving it to `missing.interpolate_1d` will, in my opinion, not work, because `missing.interpolate_1d` is not reached in the call sequence for interpolating with method `pad` since `blocks.Block.interpolate()` splits up into `blocks.Block._interpolate` (for scipy wrappers) and `blocks.Block._interpolate_with_fill` (for pad methods). `missing.interpolate_1d` is only called in the former case, i.e. for methods using the scipy wrappers.

Reply:
@cchwala interpolate_2d will work on a 1d array. did you investigate applying it along an axis (with masking logic) rather than creating a 1d version.

Reply:
see #34749 for alternative implementation calling interpolate_2d instead. interpolate_2d already has the limit logic so no need to use the preserve_nans set based logic.

Reply:
I used the nice solution from #34749 as suggested above

Reply:
as a consequence there is no `interpolate_1d_fill` anymore